### ISSUE TITLE
CkPicture not disposed

### ISSUE THREAD

**Issue by curt-weber** [flutter/flutter#142381]
### Steps to reproduce

1. Run sample in web
2. Observe log with leaks
3. CkPicture leaks have no detailed/retaining path

This leak only applies to the web platform. Tried to check the phases to see if there was already an issue for this but couldn't find it.
cc @polina-c 

### Expected results

CkPicture doesn't leak when new frames are pumped.

### Actual results

CkPicture leaks frequently, likely due to new frames as they won't leak without new frames. (sample uses CircularProgressIndicator to constantly request new frames).

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:leak_tracker/leak_tracker.dart';

void main() {
  LeakTracking.start(
    config: const LeakTrackingConfig(onLeaks: _onLeaks),
  );

  FlutterMemoryAllocations.instance.addListener(
    (event) => LeakTracking.dispatchObjectEvent(event.toMap()),
  );

  runApp(
    const MaterialApp(
      home: Center(
        child: CircularProgressIndicator(),
      ),
    ),
  );
}

Future<void> _onLeaks(LeakSummary summary) async {
  final leaks = await LeakTracking.collectLeaks();
  for (final leak in leaks.all) {
    print(leak.toYaml('', phasesAreTests: false));
  }
}
```

</details>

### Logs

<details open><summary>Logs</summary>

```console
CkPicture:
  identityHashCode: 599787274
...
leak_tracker: 262 memory leak(s): not disposed: 262, not GCed: 0, GCed late: 0

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, 3.19.0-9.0.pre.204, on Microsoft Windows [Version 10.0.19044.3803], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.3.5)  
[√] Android Studio (version 2021.3)
[√] VS Code (version 1.85.2)
[√] Connected device (3 available)
[√] Network resources
```

</details>

**Comment by polina-c** [comment]
Steps to repro use leak tracker not in a way that is not currently tested and recommended. Can you create a widget test using this guidance and check if leaks are still detected: https://github.com/dart-lang/leak_tracker/blob/main/doc/leak_tracking/DETECT.md#test-cover-with-testwidgets

**Comment by curt-weber** [comment]
Yep, I figured it was pre-emptive, I was using the live detector to locate and fix some other leaks and noticed this one likely coming from the framework. I can try to repro with testWidgets, but it says in that document you linked that web platform isn't supported yet - should I wait then? Since this leak was only observed on web. Thanks for all your great work on this!

**Comment by polina-c** [comment]
Then I suggest instead of using leak tracker (that does not support both web apps and running app), just to add temporary counting for CkPicture in your local copy: increase global counter in constructor and decrease it in `dispose`. It will give you idea how many of them are created and not disposed. How does it sound?
